quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/configuration/mappers/PropertyMappers.java

```diff
@@ -137,22 +242,73 @@ public void addAll(PropertyMapper<?>[] mappers) {
             }
         }
 
-        private void addMapperByStage(PropertyMapper<?> mapper, Map<OptionCategory, List<PropertyMapper<?>>> mappers) {
-            mappers.computeIfAbsent(mapper.getCategory(),
-                    new Function<OptionCategory, List<PropertyMapper<?>>>() {
-                        @Override
-                        public List<PropertyMapper<?>> apply(OptionCategory c) {
-                            return new ArrayList<>();
-                        }
-                    }).add(mapper);
+        private static void addMapperByStage(PropertyMapper<?> mapper, Map<OptionCategory, List<PropertyMapper<?>>> mappers) {
+            mappers.computeIfAbsent(mapper.getCategory(), c -> new ArrayList<>()).add(mapper);
+        }
+
+        public void addMapper(PropertyMapper<?> mapper) {
+            addMapper(mapper, this::add);
+        }
+
+        public void removeMapper(PropertyMapper<?> mapper) {
+            if (!mapper.getFrom().equals(mapper.getTo())) {
+                remove(mapper.getTo());
```

Comment:
The intention behind this was that we should not allow duplicates on the "transformed-to" (attribute 'to') properties. Bcs when it happens, we can have a clash of possible transformers for the property, and the result might not be deterministic. 

However, another validation would need to be added for that case, which is not present atm. I think addressing this case could be done in a follow-up task.

So, IMO, it could be handled as you suggested for now.

I'll create the follow-up issues and link them here.

Reply:
Done here: https://github.com/keycloak/keycloak/pull/25333/files#diff-a1bb1ef613469cf669070d52be8a88079005dcaed860e8d11d4231b5e2f530f7R331-R338